### PR TITLE
[SofaGeneralSimpleFem] Remove *Containers

### PR DESCRIPTION
**Pull request by fredroy**
(not the topological ones 😏)

{Length,Radius,Stiffness,Poisson}Container was created a **long time ago** to provide an "external location" where to set data per element, regardless of the component.
But: 
 - the design was flawed (hard deps, not generic at all)
 - the remaining files are just interfaces (virtual ... =0) and there is no implementation at all in the whole codebase (deleted files in the past? forgotten plugins?), so the code itself at the moment is totally useless. Actually 3 of them was even not including the necessary headers...

So the best course of action to do IMO is to delete them, and maybe find an other design later to provide means to get data per element (opposed to data for the whole set of elements)

A nice side effect is that SofaTopologicalMapping just depends on SofaBaseTopology now.
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 1]
```diff
@@ -350,10 +350,6 @@ std::map<std::string, ComponentChange> uncreatableComponents = {
     { "BeamFEMForceField", Pluginized("v20.12", "SofaGeneralSimpleFem") },
     { "HexahedralFEMForceField", Pluginized("v20.12", "SofaGeneralSimpleFem") },
     { "HexahedralFEMForceFieldAndMass", Pluginized("v20.12", "SofaGeneralSimpleFem") },
-    { "LengthContainer", Pluginized("v20.12", "SofaGeneralSimpleFem") },
-    { "PoissonContainer", Pluginized("v20.12", "SofaGeneralSimpleFem") },
-    { "RadiusContainer", Pluginized("v20.12", "SofaGeneralSimpleFem") },
-    { "StiffnessContainer", Pluginized("v20.12", "SofaGeneralSimpleFem") },
```
Comment (guparan) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp:356]: To be added as "Removed" at the end of  this file?

File: modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/StiffnessContainer.h [hunk 2]
```diff
@@ -20,17 +20,5 @@
...
-{
-public:
-    SOFA_CLASS(StiffnessContainer,sofa::core::objectmodel::BaseObject);
-
-    virtual double getStiffness(sofa::Index index) = 0;
-};
-
-} // namespace sofa::component::container
\ No newline at end of file
+#error "StiffnessContainer has been removed (PR2099)."
```
Comment (guparan) [inline modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/StiffnessContainer.h]:
```suggestion
#error "StiffnessContainer has been removed in v21.06 (PR2099)."
```

File: modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/RadiusContainer.h [hunk 3]
```diff
@@ -22,22 +22,4 @@
...
-    /// Get the radius around a given point
-    virtual double getPointRadius(sofa::Index index) = 0;
-    /// Get the radius around a given edge
-    virtual double getEdgeRadius(sofa::Index index) = 0;
-    /// Deprecated alias for getEdgeRadius
-    double getRadius(sofa::Index index) { return getEdgeRadius(index); }
-};
-
-} // namespace sofa::component::container
+#error "RadiusContainer has been removed (PR2099)."
```
Comment (guparan) [inline modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/RadiusContainer.h]:
```suggestion
#error "RadiusContainer has been removed in v21.06 (PR2099)."
```

File: modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/PoissonContainer.h [hunk 4]
```diff
@@ -20,17 +20,5 @@
...
-class PoissonContainer : public virtual sofa::core::objectmodel::BaseObject
-{
-public:
-    SOFA_CLASS(PoissonContainer,sofa::core::objectmodel::BaseObject);
-
-    virtual double getPoisson(sofa::Index index) = 0;
-};
-
-} // namespace sofa::component::container
+#error "PoissonContainer has been removed (PR2099)."
```
Comment (guparan) [inline modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/PoissonContainer.h]:
```suggestion
#error "PoissonContainer has been removed in v21.06 (PR2099)."
```

File: modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/LengthContainer.h [hunk 5]
```diff
@@ -20,17 +20,5 @@
...
-class LengthContainer : public virtual sofa::core::objectmodel::BaseObject
-{
-public:
-    SOFA_CLASS(LengthContainer,core::objectmodel::BaseObject);
-
-    virtual double getLength(sofa::Index index) = 0;
-};
-
-} // namespace sofa::component::container
+#error "LengthContainer has been removed (PR2099)."
```
Comment (guparan) [inline modules/SofaGeneralSimpleFem/src/SofaGeneralSimpleFem/LengthContainer.h]:
```suggestion
#error "LengthContainer has been removed in v21.06 (PR2099)."
```

**Review by guparan — CHANGES_REQUESTED** [review]

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp (hunk 1 above)
Comment (fredroy) [reply SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp:356]: Nope, actually these classes were never spawnable (so a mistake to put them in the "Pluginized" list)

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Pull request merged**